src/expr/src/vector_op/split_part.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::array::{BytesGuard, BytesWriter};
+use risingwave_common::error::{ErrorCode, Result, RwError};
+
+#[inline(always)]
+pub fn split_part(
+    string_expr: &str,
+    delimiter_expr: &str,
+    nth_expr: i32,
+    writer: BytesWriter,
+) -> Result<BytesGuard> {
+    let mut split = string_expr.split(delimiter_expr);
+    let nth_val = if string_expr.is_empty() {
+        // postgres: return empty string for empty input string
+        Default::default()
+    } else if delimiter_expr.is_empty() {
+        // postgres: handle empty field separator
+        //           if first or last field, return input string, else empty string
```

Comment:
* It seems the case of empty `string_expr` can be handled naturally by logic below without special treatment.
* As for empty `delimiter_expr`, the `split` iterator can be replaced by `std::iter::once(string_expr)`, but the type checker is unhappy unless we box it or use a generic helper function on `impl Iterator<Item=&str>`. So I am okay with current implementation, except that nth == 0 should return error.

Reply:
thx!
[a899543](https://github.com/singularity-data/risingwave/pull/2822/commits/a899543286d4e1eaa182d08f036d0f77bcfcadc8)